pkg/utils/json.go

```diff
@@ -17,3 +24,42 @@ func JoinPatches(patches [][]byte) []byte {
 	result = append(result, []byte("\n]")...)
 	return result
 }
+
+// MarshalPolicy accurately marshals a policy to JSON,
+// normal marshal would cause empty sub structs in
+// policy to be non nil.
+// TODO This needs to be removed. A simpler way to encode and decode Policy is needed.
+func MarshalPolicy(policy v1.ClusterPolicy) []byte {
```

Comment:
Agreed, was facing the issue before.

Do you think to change the type in the Policy struct would help? 
i.e. add a pointer to Mutation
```
type Rule struct {
	Mutation         *Mutation         `json:"mutate,omitempty"`
}
```

Reply:
Yes, it would - but this would be a big change and should be tracked in a seperate issue